Docs: final release notes for v3.1.0

mbolivar-nordic opened the pull request:
This pull request finalizes the release notes page.

There are still a few more documentation related tasks left to do before the release. These will be deferred to a subsequent PR.

mbolivar-nordic committed:
doc: releases: resolved issues for v3.1.0

Follow along with the release checklist (#43286) like this:

$ ./scripts/release/list_issues.py -o zephyrproject-rtos -r zephyr -s
2022-02-22 -f issues.txt

Then copy issues.txt into the right place in the doc, and clean up the
resulting .rst errors by properly escaping and converting the markdown
format used in github issue titles.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic committed:
scripts: add dump_bugs_pickle.py

This prints the contents of a zephyr-bugs-$DATE.pickle file in a
format similar to the output of list_issues.py.

It will be useful to have this in tree so that we can better record
the known issues at the time of a particular zephyr release.

The pickle file itself should be created using the bug snapshot
workflow defined in .github/workflows/bug_snapshot.yaml, which can be
triggered manually from this URL:

https://github.com/zephyrproject-rtos/zephyr/actions/workflows/bug_snapshot.yaml

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic committed:
doc: releases: known issues for v3.1.0

Every release notes page includes a section for known issues.

In previous releases, we have linked to a GitHub query which displays
open issues. The problem with this approach is that the information
immediately goes stale, making this section's content less than useful.

We can do better now that we have a bug snapshot workflow and a script
capable of dumping snapshot contents in the same almost-rst-format
used by list_issues.py.

Use it as follows to archive the known open bugs at time of release:

$ wget https://builds.zephyrproject.io/zephyr/bug-snapshot/zephyr-bugs-2022-06-03.pickle.xz
$ unxz zephyr-bugs-2022-06-03.pickle.xz
$ ./scripts/dump_bugs_pickle.py zephyr-bugs-2022-06-03.pickle

Then copy paste the output into the release notes and handle .rst
escapes in the same way has previously been done for closed issues
since the last release.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic committed:
doc: releases: fixes for v3.1.0

Make a pass through the file, trying to fix as many issues as possible
with how the notes are written. This includes fixes for tense and
invalid .rst role usage, as well as removal of any sections which have
missing content.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic committed:
doc: releases: fix location of "new APIs" content for v3.1.0

This needs to be in the "API changes" section, instead of in a
"Bluetooth" section where it currently is.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic committed:
doc: releases: unite disjoint bluetooth sections for v3.1.0

For some reason, there are two different sections covering bluetooth.
This is clearly incorrect; fix it by moving the content into a single
section.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic committed:
doc: releases: unify disjoint issue related sections for v3.1.0

There are three lists of issues in the release notes:

- security vulnerabilities which received CVEs
- other open bugs
- resolved issues

(Here, "issue" is a strictly more general term than "bug": every bug
is an issue, not every issue is a bug. This is GitHub-specific
jargon.)

The current structure is awkward. Clean it up by unifying the
lists under a single top level section and cleaning up the distinction
between issues and bugs.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic commented:
We could use someone with superpowers to override the compliance failures. Those are copied straight from the github issue titles themselves.

dkalowsk commented on doc/releases/release-notes-3.1.rst:
```diff
@@ -1123,3 +1123,799 @@ Issue Related Items
+* :github:`46212` - Release notes for v3.1.0: dma
+* :github:`46211` - Release notes for v3.1.0: disk
+* :github:`46210` - Release notes for v3.1.0: arm64
+* :github:`46209` - Release notes for v3.1.0: counter
```
do these release note piece need to be in the release notes issue list?  Seems like it's not really of interest for an end user to see such administravia.

dkalowsk commented on doc/releases/release-notes-3.1.rst:
```diff
@@ -1123,3 +1123,799 @@ Issue Related Items
+* :github:`45883` - Bluetooth: Controller: CCM reads data before Radio stores them when DF enabled on PHY 1M
+* :github:`45882` - Zephyr minimal C library contains files licensed with BSD-4-Clause-UC
+* :github:`45878` - doc: release: Update release notes with CVE
+* :github:`45877` - version bump to 2.7.1
```
There are a lot of 2.7 branch details in here.  Are they needed for 3.1 release notes?

dkalowsk commented on doc/releases/release-notes-3.1.rst:
```diff
@@ -1123,3 +1123,799 @@ Issue Related Items
+* :github:`45455` - [backport v2.7-auditable-branch] backport of #45400 failed
+* :github:`45454` - [backport v2.7-branch] backport of #45400 failed
+* :github:`45451` - [backport v3.0-branch] backport of #43845 failed
+* :github:`45448` - Request for backporting to 3.0.0
```
looks like a few for 3.0 backporting as well

dkalowsk commented on doc/releases/release-notes-3.1.rst:
More backport request

dkalowsk commented on doc/releases/release-notes-3.1.rst:
This issue is ... not clear/useful

mbolivar-nordic replied:
I'll remove this one since it was converted to a discussion.

dkalowsk commented on doc/releases/release-notes-3.1.rst:
again missing some details here

mbolivar-nordic replied:
That's the reporter's fault; I'm OK with not fixing it for these release notes.

dkalowsk reviewed (dismissed):
Marking as approved as you may be okay with these bits in there.  I don't believe they add any value to the release notes, but I'll leave that to you to make the call.  Everything else looks good to the PR.

mbolivar-nordic committed:
doc: release: manually curate list_issues.py output

Apply suggestions from Dan Kalowsky to clean up extraneous issues
included in the script's output.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic committed:
doc: release: 3.1.0 notes are no longer a working draft

Remove that from the title.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic commented:
@dkalowsk feedback addressed, thank you. I

mbolivar-nordic commented:
Fixed a doc build error; not sure why I didn't notice that in my local builds.
